packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -252,7 +252,14 @@ class PageController extends ScrollController {
   void attach(ScrollPosition position) {
     super.attach(position);
     final _PagePosition pagePosition = position;
-    pagePosition.viewportFraction = viewportFraction;
+
+    if (pagePosition.viewportFraction == viewportFraction)
```

Comment:
from `ScrollPosition.forcePixel` documentation:
>   /// This should not be called during layout (e.g. when setting the initial
  /// scroll offset). Consider [correctPixels] if you find you need to adjust
  /// the position during layout.

Also it doesn't sound right to notify the listeners in `ScrollController.attach`.

Reply:
Why move it out of the setter though?

Reply:
IMO since using either `forcePixel` or `correctPixel` needs to be decided on a case-by-case basis, it feels a bit error-prone to have a `viewportFraction` setter which sounds like a silver bullet for changing `viewportFraction`.
Also `attach` is the only caller of the setter right now, so I got rid of the setter and moved the implementation to `attach`.

Reply:
this is build, not layout. should be fine to call during build.